### PR TITLE
Add Linux support

### PR DESCRIPTION
**Pull request by foad**
Adds support for attaching to the appium socket on Linux

 - Code is pretty much a copy/paste of the YiMac one with all of the Xcode-related bits removed
 - Random lint changes came about after I ran `npm run lint -- --fix` so I guess they should be fine?
 - Wasn't sure on whether to call it YiLinux or just Linux so if I got it wrong please let me know 🙏 
 - Have tested locally by using `npm link` in a clone of `appium` and it works as expected on our Ubuntu 20.04 VMs

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -158,7 +169,7 @@ class YouiEngineDriver extends BaseDriver {
     if (this.caps.platformName !== null) {
       let appPlatform = this.caps.platformName.toLowerCase();
 
-      if (['yimac', 'yitvos', 'bluesky'].includes(appPlatform)) {
+      if (['yimac', 'yitvos', 'bluesky', 'yilinux'].includes(appPlatform)) {
```
Comment (lukeyoui) [inline lib/driver.js:172]: It is fine to call it yilinux since it is a custom platform for our driver

File: lib/driver.js [hunk 2]
```diff
@@ -423,16 +442,17 @@ class YouiEngineDriver extends BaseDriver {
         errno = 'EOK';
       }
 
-      if (!connected && retryCount === (MAX_RETRY_COUNT - 1)) {
-        logger.errorAndThrow('Failed to connect ' + MAX_RETRY_COUNT + ' times. Aborting.');
+      if (!connected && retryCount === MAX_RETRY_COUNT - 1) {
+        logger.errorAndThrow(
+          'Failed to connect ' + MAX_RETRY_COUNT + ' times. Aborting.'
+        );
```
Comment (lukeyoui) [inline lib/driver.js:448]: I assume removing the brackets doesn't functionally change anything, but it did make it more readable. Was this flagged by lint?

File: lib/driver.js (hunk 1 above)
Comment (foad) [reply lib/driver.js:172]: Thanks for confirming 🙏

File: lib/driver.js (hunk 2 above)
Comment (foad) [reply lib/driver.js:448]: Yeah linter did this automatically when I ran `npm run lint -- --fix`

**Comment by foad** [comment]
@lukeyoui 

Thank you

Yeah I built for linux via the following:

```
> generate.rb -p linux --inline --file index.youi.js -c Release -d YI_APPIUM="ON"
> build.rb -b build/linux/Release -c Release
```

And then just connected the same way we do for other platforms

Want me to update the README in this PR?

**Comment by lukeyoui** [comment]
@foad 

Thanks, I'll see if I can get my hands on a linux environment to give it a quick run through.

That would be great if you could update the README in this PR.

**Comment by foad** [comment]
@lukeyoui Should I be bumping the package version in this PR or would that be done after this is merged?

**Comment by lukeyoui** [comment]
@foad Thanks for updating the readme. I will update the package version after we merge the PR once I go to publish, that way we only have to do it once.

**Pull request merged**